File: app/src/main/java/com/friendlypos/reimpresion/fragment/ReimprimirFacturaFragment.java
package com.friendlypos.reimpresion.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.friendlypos.R;
import com.friendlypos.distribucion.fragment.BaseFragment;
import com.friendlypos.distribucion.modelo.sale;
import com.friendlypos.reimpresion.activity.ReimprimirActivity;
import com.friendlypos.reimpresion.adapters.ReimprimirFacturaAdapter;

import java.util.List;

import butterknife.Bind;
import butterknife.ButterKnife;
import io.realm.Realm;
import io.realm.RealmQuery;
import io.realm.RealmResults;

public class ReimprimirFacturaFragment extends BaseFragment {

    @Bind(R.id.recyclerViewReimprimirFactura)
    public RecyclerView recyclerView;

    private ReimprimirFacturaAdapter adapter;

    RealmResults<sale> result1;
    Realm realm;

    public static ReimprimirFacturaFragment getInstance() {
        return new ReimprimirFacturaFragment();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

    @Override
    public void onResume() {
        super.onResume();
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_reimprimir_factura, container,
                false);
        ButterKnife.bind(this, rootView);
        return rootView;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        adapter = new ReimprimirFacturaAdapter(getContext(), ((ReimprimirActivity)getActivity()), getListClientes());
        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        recyclerView.setAdapter(adapter);


    }

    private List<sale> getListClientes(){
        realm = Realm.getDefaultInstance();
        RealmQuery<sale> query = realm.where(sale.class).equalTo("aplicada", 1).equalTo("facturaDePreventa", "Distribucion").or().equalTo("facturaDePreventa", "VentaDirecta");
        RealmResults<sale> result1 = query.findAll();

        return result1;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        realm.close();
    }

    @Override
    public void updateData() {

    }
}

File: app/src/main/java/com/friendlypos/reimpresion/fragment/ReimprimirFacturaFragment.kt
package com.friendlysystemgroup.friendlypos.reimpresion.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import com.friendlysystemgroup.friendlypos.databinding.FragmentReimprimirFacturaBinding
import com.friendlysystemgroup.friendlypos.distribucion.fragment.BaseFragment
import com.friendlysystemgroup.friendlypos.distribucion.modelo.sale
import com.friendlysystemgroup.friendlypos.reimpresion.activity.ReimprimirActivity
import com.friendlysystemgroup.friendlypos.reimpresion.adapters.ReimprimirFacturaAdapter
import io.realm.Realm
import io.realm.Sort

/**
 * Fragmento para seleccionar una factura a reimprimir
 */
class ReimprimirFacturaFragment : BaseFragment() {
    private var _binding: FragmentReimprimirFacturaBinding? = null
    private val binding get() = _binding!!
    
    private var adapter: ReimprimirFacturaAdapter? = null
    private var realm: Realm? = null

    override fun onCreateView(
        inflater: LayoutInflater, 
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentReimprimirFacturaBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupRecyclerView()
    }
    
    private fun setupRecyclerView() {
        val activity = activity as? ReimprimirActivity
        activity?.let {
            val facturas = obtenerListaFacturas()
            
            adapter = ReimprimirFacturaAdapter(
                requireContext(), 
                it,
                facturas
            )
            
            binding.recyclerViewReimprimirFactura.apply {
                setHasFixedSize(true)
                layoutManager = LinearLayoutManager(requireActivity())
                adapter = this@ReimprimirFacturaFragment.adapter
            }
        }
    }

    /**
     * Obtiene la lista de facturas para mostrar
     */
    private fun obtenerListaFacturas(): List<sale> {
        realm = Realm.getDefaultInstance()
        
        return realm?.let { realm ->
            realm.where(sale::class.java)
                .equalTo("aplicada", 1)
                .beginGroup()
                    .equalTo("facturaDePreventa", "Distribucion")
                    .or()
                    .equalTo("facturaDePreventa", "VentaDirecta")
                .endGroup()
                .findAllSorted("id", Sort.DESCENDING)
                .toList()
        } ?: emptyList()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
    
    override fun onDestroy() {
        super.onDestroy()
        realm?.close()
        realm = null
    }

    override fun updateData() {
        // Actualizar datos cuando cambia la pestaña
        adapter?.let { adapter ->
            adapter.contentList = obtenerListaFacturas()
            adapter.notifyDataSetChanged()
        }
    }

    companion object {
        fun newInstance(): ReimprimirFacturaFragment = ReimprimirFacturaFragment()
    }
}
